server/middlewares/validators/videos/video-channel-sync.ts

```diff
@@ -0,0 +1,75 @@
+import * as express from 'express'
+import { isUrlValid } from "@server/helpers/custom-validators/activitypub/misc"
+import { logger } from "@server/helpers/logger"
+import { body, param } from "express-validator"
+import { areValidationErrors, doesVideoChannelIdExist } from "../shared"
+import { HttpStatusCode, VideoChannelSyncCreate } from '@shared/models'
+import { VideoChannelSyncModel } from '@server/models/video/video-channel-sync'
+import { VideoChannelModel } from '@server/models/video/video-channel'
+import { CONFIG } from '@server/initializers/config'
+
+export const ensureSyncIsEnabled = (req: express.Request, res: express.Response, next: express.NextFunction) => {
+  if (!CONFIG.IMPORT.VIDEOS.HTTP.ENABLED) {
+    return res.fail({
+      status: HttpStatusCode.FORBIDDEN_403,
+      message: 'Synchronization is impossible as video upload via HTTP is not enabled on the server'
+    })
+  }
+  return next()
+}
+
+export const videoChannelSyncValidator = [
+  body('externalChannelUrl').custom(isUrlValid).withMessage('Should have a valid channel url'),
+  body('videoChannelId').isInt().withMessage('Should have a valid video channel id'),
+
+  async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+    logger.debug('Checking videoChannelsAdd parameters', { parameters: req.body })
+
+    if (areValidationErrors(req, res)) return
+
+    const body: VideoChannelSyncCreate = req.body
+    if (!await doesVideoChannelIdExist(body.videoChannelId, res)) {
+      return
+    }
+
+    return next()
+  }
+]
+
+export const ensureSyncExists = [
+  async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+
+    const syncId = parseInt(req.params.id, 10)
+    const sync = await VideoChannelSyncModel.loadById(syncId)
+    if (!sync) {
+      return res.fail({
+        status: HttpStatusCode.NOT_FOUND_404,
+        message: 'Synchronization not found'
+      })
+    }
+    res.locals.videoChannelSync = sync
+    return next()
+  }
+]
+
+export const ensureSyncTargetChannelExists = [
+  async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+    const sync = res.locals.videoChannelSync
```

Comment:
Since we don't allow `null` for channels, I think the `load` can be renamed `loadAndPopulateChannel` that automatically fetches the entry with the channel.